docs/rfcs/XeGPU.md

```diff
@@ -701,6 +700,332 @@ An example on how to perform transpose using load with chunk_size in SIMT flavor
+
+When a wg_map attribute is attached to a tensor descriptor, load/store/dpas will operate at the workgroup level. The wg_map attribute must be specified when creating the tensor descriptor.
+
+**Constraints**
+
+Given these definitions:
+```mlir
+sg_data_size = sg_data[0] × sg_data[1]
+workgroup_size = sg_layout[0] × sg_layout[1]
+tensor_size = tensor_desc[0] × tensor_desc[1]
+```
+
+The following conditions must hold:
+
+* workgroup_size must represent the number of subgroups in a workgroup for a kernel.
+* tensor_desc[0] must be either evenly divisible by sg_layout[0] × sg_data[0], or vice versa.
+* tensor_desc[1] must be either evenly divisible by sg_layout[1] × sg_data[1], or vice versa.
+
+**distribution rule**
+
+The tensor_desc is distributed to sg_data x sg_layout along each dimension in a round-robin fashion. If sg_data[i] x sg_layout[i] < tensor_desc[i], we have data left after all subgroups are assigned for the first round, we continue to assign the rest data starting from the first subgroup until the data is completely assigned. If sg_data[i] x sg_layout[i] > tensor_desc[i], we may have already used up all the data before all subgroups are assigned. In this case, we wrap around the tensor data and continue the assignment, and the rest subgroups along that dimension share the same data.
+
+**Resulting WI Data Fragment**
+
+The distributed tensor for each subgroup has the same dimension as the work group level tensor. 
+
+**Examples of workgroup distribution with wg_map**
+
+The workgroup creates a tensor descriptor [128, 128] and distributes to 4 subgroups with `sg_layout` [2,2], and each subgroup gets `sg_data` [32,128]. The first dimension is split and distributed to subgroups in two rounds, and the second dimension is assigned as whole to multiple subgroup threads.
+
+```mlir
+   #wg_map_a = #xegpu.wg_map<sg_layout = [2, 2], sg_data = [32, 128], sg_order = [1, 0]>
+   %wg_tdesc = xegpu.create_nd_tdesc %A[%m, %c0] : memref<1024x1024xf16> -> tensor_desc<128x128xf16, #wg_map_a>
+```
+The table below shows the result tensor for each subgroup thread and its linear subgroup thread id.
+
+| subgroup tensor	| 2D subgroup id	| Linearized subgroup id
+| :---   | :----   | :----   |
+| [ 0:31, 0:127] | [0, 0], [0, 1] | 0 , 1 |
+| [ 32:63, 0:127] | [1, 0], [1, 1] | 2 , 3 |
+| [ 64:95, 0:127] | [0, 0], [0, 1] | 0 , 1 |
+| [ 96:127, 0:127] | [1, 0], [1, 1] | 2 , 3 |
+
+The `wg_map` attribute propagates from the matrix multiplication ops to other ops. Since we can't attatch the `wg_map` attribute to MLIR vector data type, we attach the attribute to vector type-based operations temporarily within the workgroup distribution pass. The `wg_map` attribute propagation can be performed from output to input, or the other direction. We describes below the propagation rules from output to input for typical operations including dpas, reduction, broadcast, shape_cast, and transpose.
```

Comment:
This assumes a gemm kernel. We might want to consider layout anchor operation that we can propagate from (e.g. `xegpu.assignLayout %vec, {#wg_map_a}`). This would cover both gemm and non-gemm cases, then we can remove the layout out of tensor_desc data type.

Reply:
store is another anchor. would that be good enough for non-gemm use case?

Reply:
What I meant is a dedicated instruction with the layout attribute part of the op definition, so it the attribute cannot be dropped by MLIR folding passes.

Reply:
What is the specific anchor operations on non-gemm case you are looking for? 
One alternative is to extend convert_layout op, which accepts identical map at the beginning of the propagation, to indicate the requirement of layout. But better to understand the use case better. 
%vector_a = xegpu.convert_layout %vector_b {#sg_map_a #sg_map_a }: vector<256x256xfloat> into vector<256x256xfloat>